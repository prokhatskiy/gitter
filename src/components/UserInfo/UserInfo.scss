.user-info {
  border: 1px solid #000;
  margin: 0 0 30px;
  text-align: center;
  padding: 10px;
  height: 220px;

  &.loading {
    & > :not(.loader) {
      display: none;
    }
  }
}