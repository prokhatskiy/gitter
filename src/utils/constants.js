export const ROUTES = {
  MAIN: '/',
  LOGIN: '/login',
  DEFAULT: '*'
};

export const API = {
  LOGIN: '/auth/login',
  LOGOUT: '/auth/logout',
  USER: '/api/user',
  ROOMS: '/api/rooms',
  MESSAGES: '/api/messages',
  STREAM: '/api/subscribe/messages/'
};