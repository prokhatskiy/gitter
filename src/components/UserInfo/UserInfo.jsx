import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import './UserInfo.scss';

import { API } from 'Utils/constants';

class UserInfo extends Component {
  static propTypes = {
    user: PropTypes.shape({
      avatarUrlMedium: PropTypes.string,
      displayName: PropTypes.string
    }),
    fetchUser: PropTypes.func
  };

  static defaultProps = {
    user: {},
    fetchUser: () => {}
  };

  componentWillMount() {
    this.props.fetchUser();
  }
  render() {
    const { user: { avatarUrlMedium, displayName, status } } = this.props;
    const isLoading = status === 'load';

    const cn = cx('user-info', {
      loading: isLoading
    });

    return (
      <div className={cn}>
        { isLoading && <span className="loader">Loading...</span>}
        {
          avatarUrlMedium &&
          <img src={avatarUrlMedium} alt={displayName} className="user-info__pic" />
        }
        <h1 className="user-info__name">
          {displayName}
        </h1>
        <a href={API.LOGOUT} className="user-info__logout">Logout</a>
      </div>
    );
  }
}

export default UserInfo;