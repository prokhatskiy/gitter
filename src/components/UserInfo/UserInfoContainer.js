import { connect } from 'react-redux';
import { fetchUser } from 'Redux/actions';
import { getUser } from 'Redux/selectors';

import UserInfo from './UserInfo';

export const mapDispatchToProps = {
  fetchUser
};

export default connect(getUser, mapDispatchToProps)(UserInfo);