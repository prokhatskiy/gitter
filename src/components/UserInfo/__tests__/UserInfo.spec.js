import React from 'react';
import { shallow } from 'enzyme';

import UserInfo from '../UserInfo';
import { API } from 'Utils/constants';

describe('<UserInfo /> component', () => {
  it('should be rendered without props', () => {
    const wrapper = shallow(<UserInfo />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render loader and have appropriate className', () => {
    const user = {
      status: 'load'
    };

    const wrapper = shallow(<UserInfo user={user} />);

    expect(wrapper.contains(<span className="loader">Loading...</span>)).toEqual(true);
    expect(wrapper.hasClass('loading')).toEqual(true);
  });

  it('should fetch user info when mounted', () => {
    const fetchUserMock = jest.fn();

    shallow(<UserInfo fetchUser={fetchUserMock} />);

    expect(fetchUserMock).toHaveBeenCalledTimes(1);
  });

  it('should contain logout link', () => {
    const user = {
      status: 'fetched',
      displayName: 'Artem'
    };

    const wrapper = shallow(<UserInfo user={user} />);

    expect(wrapper.find('a').prop('href')).toEqual(API.LOGOUT);
  });

  it('should render image if avatar url is presented', () => {
    const user = {
      status: 'fetched',
      displayName: 'Artem'
    };

    let wrapper = shallow(<UserInfo user={user} />);

    expect(wrapper.find('img').exists()).toEqual(false);

    const userWithPic = {
      ...user,
      avatarUrlMedium: 'test'
    };

    wrapper = shallow(<UserInfo user={userWithPic} />);

    const img = wrapper.find('img');

    expect(img.exists()).toEqual(true);
    expect(img.prop('src')).toEqual(userWithPic.avatarUrlMedium);
  });
});