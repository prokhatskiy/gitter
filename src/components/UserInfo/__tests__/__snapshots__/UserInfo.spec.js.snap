// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> component should be rendered without props 1`] = `
<div
  className="user-info"
>
  <h1
    className="user-info__name"
  />
  <a
    className="user-info__logout"
    href="/auth/logout"
  >
    Logout
  </a>
</div>
`;
